src/main/java/org/opensearch/ad/settings/AnomalyDetectorSettings.java

```diff
@@ -222,6 +222,21 @@ private AnomalyDetectorSettings() {}
             Setting.Property.Dynamic
         );
 
+    // for a batch operation, we want all of the bounding box in-place for speed
+    public static final double BATCH_BOUNDING_BOX_CACHE_RATIO = 1;
+
+    // for a real-time operation, we trade off speed for memory as real time opearation
+    // only has to do one update/scoring per interval
+    public static final double REAL_TIME_BOUNDING_BOX_CACHE_RATIO = 0;
+
+    public static final int DEFAULT_SHINGLE_SIZE = 8;
+
+    // Multi-entity detector model setting:
+    // TODO (kaituo): change to 4
```

Comment:
This TODO means we need to change this hard coded HCAD shingle size from 1 to 4? Or we should set the HCAD shingle size upper limit as 4 (or some other value)?

Reply:
It means we should have default value 4 but allow people to change like single-stream detectors.

Reply:
Got it. Should we set different upper limit of shingle size for single-stream detector and HCAD?

Reply:
We can do that when there is a need.

Reply:
sure